### PR TITLE
AMP form security improvements

### PR DESCRIPTION
**Pull request by kul3r4**
- Removes action where POST method is required
- improves samples to use GET only for info retrieval and search
This addresses #416, we still need to work on improving the shopping cart implementation for the Product sample, but will do in a separate PR.

@sebastianbenz PTAL
@rowan-m FYI In the Hotel sample,  I moved POST -> GET  for Check availability and it seems it is an info retrieval. I added an XHR endpoint for the BOOK button given the action is a POST.

### CODE REVIEW

File: src/20_Components/amp-form.html [hunk 1]
```diff
@@ -75,7 +76,7 @@
     The `amp-form` component displays `submit-success` or `submit-error` elements based on the response and
     renders the template data inside these two elements.
   -->
-  <form method="post" action="<%host%>/components/amp-form/submit-form-input-text" action-xhr="<%host%>/components/amp-form/submit-form-input-text-xhr"  target="_top">
+  <form method="post" action-xhr="<%host%>/components/amp-form/submit-form-input-text-xhr"  target="_top">
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: action attribute is required. applies to other changes as well.

Comment (kul3r4) [inline src/20_Components/amp-form.html]: Hi, look at [validator](https://github.com/ampproject/amphtml/blob/553247b6a408a06e144e05eeb7dcfa96bfce0335/src/document-submit.js), it seems that if it's not a GET, action shouldn't be specified.

Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: Hm - I was looking at the documentation. 

@mkhatib: POST + action - who is right? Documentation or validator?

Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: @kul3r4 you're right. In that case we can get rid of target as well.

File: src/20_Components/amp-form.html [hunk 2]
```diff
@@ -102,7 +103,7 @@
   }
   ```
 -->
-<form class="hide-inputs" method="post" action="<%host%>/components/amp-form/submit-form-input-text" action-xhr="<%host%>/components/amp-form/submit-form-input-text-xhr" target="_top">
+<form class="hide-inputs" method="post" action-xhr="<%host%>/components/amp-form/submit-form-input-text-xhr" target="_top">
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: remove target (and for all the following samples)

File: src/20_Components/amp-form.html [hunk 3]
```diff
@@ -346,27 +347,11 @@
...
-        Success! Thanks for trying the <code>amp-form</code> demo!
-      </template>
-    </div>
-    <div submit-error>
-      <template type="amp-mustache">
-        Error!
-      </template>
-    </div>
-  </form>
+
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: why did you remove this sample?

File: src/20_Components/amp-form.html [hunk 4]
```diff
@@ -55,14 +55,15 @@
   <!-- #### Form Submission with Page Reload -->
   <!--
     Use the `action` attribute to specify a server-endpoint which should handle the form input. The URL must be HTTPS.
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: Add: "This works only for `GET` requests.

File: src/20_Components/amp-form.html (hunk 3 above)
Comment (kul3r4) [inline src/20_Components/amp-form.html]: Because search doesn't make sense with POST. Line 64 contains search with GET.

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

File: src/20_Components/amp-form.html (hunk 3 above)
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: my fault
